jsvg-javafx/src/main/java/com/github/weisj/jsvg/ui/jfx/FXSVGCanvas.java

```diff
@@ -0,0 +1,301 @@
+    }
+
+    ////////////////////////////////////////////////
+
+    private final BooleanProperty useSVGViewBox = new SimpleBooleanProperty(DEFAULT_USE_SVG_VIEW_BOX);
+
+    public boolean isUsingSVGViewBox() {
+        return useSVGViewBox.get();
+    }
+
+    public BooleanProperty useSVGViewBoxProperty() {
+        return useSVGViewBox;
+    }
+
+    public void setUseSVGViewBox(boolean useSVGViewBox) {
+        this.useSVGViewBox.set(useSVGViewBox);
+    }
+
+    ////////////////////////////////////////////////
+
+    private final ObjectProperty<@Nullable ViewBox> currentViewBox = new SimpleObjectProperty<>();
+
+    public @Nullable ViewBox getCurrentViewBox() {
+        return currentViewBox.get();
+    }
+
+    public @Nullable ReadOnlyObjectProperty<@Nullable ViewBox> currentViewBoxProperty() {
+        return currentViewBox;
+    }
+
+    ////////////////////////////////////////////////
+
+    private final ObjectProperty<@Nullable ViewBox> viewBox = new SimpleObjectProperty<>(DEFAULT_VIEW_BOX);
+
+    public @Nullable ViewBox getViewBox() {
+        return viewBox.get();
+    }
+
+    public @NotNull ObjectProperty<@Nullable ViewBox> viewBoxProperty() {
+        return viewBox;
+    }
+
+    public void setViewBox(@Nullable ViewBox viewBox) {
+        this.viewBox.set(viewBox);
+    }
+
+    ////////////////////////////////////////////////
+
+    private final BooleanProperty animated = new SimpleBooleanProperty(DEFAULT_ANIMATED);
+
+    public boolean isAnimated() {
+        return animated.get();
+    }
+
+    public @NotNull BooleanProperty animatedProperty() {
+        return animated;
+    }
+
+    public void setAnimated(boolean animated) {
+        this.animated.set(animated);
+    }
+
+    ////////////////////////////////////////////////
+
+    private final LongProperty animationElapsedTime = new SimpleLongProperty();
+
+    public long getAnimationElapsedTime() {
+        return animationElapsedTime.get();
+    }
+
+    public @NotNull ReadOnlyLongProperty animationElapsedTimeProperty() {
+        return animationElapsedTime;
+    }
+
+    ////////////////////////////////////////////////
+
+    private final ObjectProperty<@NotNull Animation> currentAnimation = new SimpleObjectProperty<>(DEFAULT_ANIMATION);
+
+    public @NotNull Animation getCurrentAnimation() {
+        return currentAnimation.get();
+    }
+
+    public @NotNull ReadOnlyObjectProperty<@NotNull Animation> currentAnimationProperty() {
+        return currentAnimation;
+    }
+
+    ////////////////////////////////////////////////
+
+    private final ObjectProperty<@Nullable Animation> animation = new SimpleObjectProperty<>();
+
+    public @Nullable Animation getAnimation() {
+        return animation.get();
+    }
+
+    public @NotNull ObjectProperty<@Nullable Animation> animationProperty() {
+        return animation;
+    }
+
+    public void setAnimation(@Nullable Animation animation) {
+        this.animation.set(animation);
+    }
```

Comment:
What is the difference between these two?

Reply:
It was originally a way to access the NonNull animation + viewbox, the thinking was it might be useful for situations like providing a player interface with start + stop and showing the duration of the animation without having to access the document. I've now removed these though as I think they are more confusing than useful as it only covers a very specific use case.